### PR TITLE
feat(EMS-2366): No PDF - Your Buyer - Connection to the buyer page

### PR DESCRIPTION
**Pull request by Zainzzkk**
## Introduction :pencil2:
This PR adds the connection to the buyer page

## Resolution :heavy_check_mark:
- Added connection to the buyer page and routes
- Added unit and e2e test

## Miscellaneous :heavy_plus_sign:
- Added shared `single-radio-page` template

### CODE REVIEW

File: e2e-tests/insurance/cypress/e2e/journeys/your-buyer/connection-to-the-buyer/connection-to-the-buyer-page.spec.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+const {
+  YOUR_BUYER: { COMPANY_OR_ORGANISATION, CONNECTION_TO_THE_BUYER, WORKING_WITH_BUYER },
+} = ROUTES.INSURANCE;
+
+const FIELDS = YOUR_BUYER_FIELDS.WORKING_WITH_BUYER;
+
+const baseUrl = Cypress.config('baseUrl');
+
+context('Insurance - Your Buyer - Connection with the buyer - As an exporter, I want to provide the details on trading history with the buyer of my export trade, So that UKEF can gain clarity on whether I have  trading history with the buyer as part of due diligence', () => {
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/your-buyer/connection-to-the-buyer/connection-to-the-buyer-page.spec.js:27]: The description mentiones trading history with the buyer?

Comment (Zainzzkk) [reply e2e-tests/insurance/cypress/e2e/journeys/your-buyer/connection-to-the-buyer/connection-to-the-buyer-page.spec.js:27]: thats come from the ticket

File: e2e-tests/insurance/cypress/e2e/journeys/your-buyer/connection-to-the-buyer/connection-to-the-buyer-page.spec.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        it('should not render a label', () => {
+          field(fieldId).label().should('not.be.visible');
+        });
+
+        it('should not render a visible input', () => {
+          field(fieldId).textarea().should('not.be.visible');
+        });
+      });
+
+      describe(`${CONNECTED_WITH_BUYER} yes radio selected`, () => {
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/your-buyer/connection-to-the-buyer/connection-to-the-buyer-page.spec.js:107]: Should we/could we align with `about-goods-or-services.spec.js` ? e.g it's more like a story

```js
describe(`when clicking the 'yes' ${FINAL_DESTINATION_KNOWN} radio`, () => { ...
```

File: e2e-tests/insurance/cypress/e2e/journeys/your-buyer/connection-to-the-buyer/validation/connection-to-the-buyer-validation.spec.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+const {
+  ROOT: INSURANCE_ROOT,
+  YOUR_BUYER: { CONNECTION_TO_THE_BUYER },
+} = INSURANCE_ROUTES;
+
+const { MAXIMUM } = YOUR_BUYER_FIELDS.WORKING_WITH_BUYER[CONNECTION_WITH_BUYER_DESCRIPTION];
+
+const baseUrl = Cypress.config('baseUrl');
+
+context('Insurance - Your Buyer - Connection to the buyer page - Page validation', () => {
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/your-buyer/connection-to-the-buyer/validation/connection-to-the-buyer-validation.spec.js:31]: form validation 😛

File: src/ui/server/controllers/insurance/your-buyer/connection-to-the-buyer/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+      it('should render template with validation errors', async () => {
+        await post(req, res);
+
+        const payload = constructPayload(req.body, FIELD_IDS);
+
+        const validationErrors = generateValidationErrors(payload);
+
+        const expectedVariables = {
+          ...insuranceCorePageVariables({
+            PAGE_CONTENT_STRINGS: PAGES.INSURANCE.YOUR_BUYER.CONNECTION_TO_THE_BUYER,
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/your-buyer/connection-to-the-buyer/index.test.ts:135]: `PAGE_CONTENT_STRINGS` const and individual unit test?

File: src/ui/server/controllers/insurance/your-buyer/connection-to-the-buyer/index.test.ts [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+            BACK_LINK: req.headers.referer,
+          }),
+          ...PAGE_VARIABLES,
+          FIELD_HINT: PAGE_VARIABLES.PAGE_CONTENT_STRINGS.HINT,
+          userName: getUserNameFromSession(req.session.user),
+          validationErrors,
+          submittedValues: payload,
+          SAVE_AND_BACK_URL: `${INSURANCE_ROOT}/${mockApplication.referenceNumber}${SAVE_AND_BACK}`,
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/your-buyer/connection-to-the-buyer/index.test.ts:143]: `SAVE_AND_BACK_URL` could be in `pageVariables`, DRY

File: src/ui/server/controllers/insurance/your-buyer/connection-to-the-buyer/index.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+export const FIELD_ID = CONNECTED_WITH_BUYER;
+
+export const FIELD_IDS = [CONNECTED_WITH_BUYER, CONNECTION_WITH_BUYER_DESCRIPTION];
+
+export const TEMPLATE = TEMPLATES.INSURANCE.YOUR_BUYER.CONNECTION_TO_THE_BUYER;
+
+export const PAGE_VARIABLES = {
+  FIELD_ID,
+  FIELDS: {
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/your-buyer/connection-to-the-buyer/index.ts:29]: Why do we have `FIELD_ID` and `FIELDS`?

File: src/ui/server/controllers/insurance/your-buyer/connection-to-the-buyer/validation/rules/index.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import connectedToBuyerRule from './connected-to-buyer';
+import tradedWithBuyerRule from './connection-with-buyer-description';
+import { ValidationErrors } from '../../../../../../../types';
+
+const rules = [connectedToBuyerRule, tradedWithBuyerRule] as Array<() => ValidationErrors>;
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/your-buyer/connection-to-the-buyer/validation/rules/index.ts:5]: `tradedWithBuyerRule` typo

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**